pkg/transformers/json/README.md

```diff
@@ -2,3 +2,53 @@
 
 JSON Transformer provides Message Transformer for JSON messages.
 To transform Mainflux Message successfully, the payload must be a JSON object.
+
+For the messages that contain _JSON array as the root element_, JSON Transformer does normalization of the data: it creates a separate JSON message for each JSON object in the root. In order to be processed and stored properly, JSON messages need to contain message format information. For the sake of the simpler storing of the messages, nested JSON objects are flatten to a single JSON object, using composite keys with the default separator `/`. This implies that the separator character (`/`) _is not allowed in the JSON object key_. For example, the following JSON object:
+```json
+{
+    "name": "name",
+    "id":8659456789564231564,
+    "in": 3.145,
+    "alarm": true,
+    "ts": 1571259850000,
+    "d": {
+        "tmp": 2.564,
+        "hmd": 87,
+        "loc": {
+            "x": 1,
+            "y": 2
+        }
+    }
+}
+```
+
+will be transformed to:
+
+```json
+
+{
+    "name": "name",
+    "id":8659456789564231564,
+    "in": 3.145,
+    "alarm": true,
+    "ts": 1571259850000,
+    "d/tmp": 2.564,
```

Comment:
Why do we use `/` instead of `.` for flattening?

Reply:
The only reason is that it seems to be more likely that `.` is already a part of the key. Basically, we can choose any character to be the delimiter, and that's the only character that _can't be in any key in the received JSON object_. It can be some of the usual delimiters: `,`, `/`, `\` `.`, `|`, or special chars such as `#`, `?`, `!`, `>`...